### PR TITLE
Update rails console prompt

### PR DESCRIPTION
**Pull request by st0012**
### Motivation / Background

This PR adopts the follow up suggestion in https://github.com/rails/rails/issues/50770 around Rails console's prompt.

### Detail

Instead of displaying `[env]:[line-number]>` as the prompt, showing `[app-name]([env])>` will provide more value to users.

For example, if the app's top-level module is `MyApp`, its development prompt would be:

```
my-app(dev)>
```

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: railties/lib/rails/commands/console/console_command.rb
Comment (byroot) [inline railties/lib/rails/commands/console/console_command.rb:38]: Nitpick: 
```suggestion
        app_name = @app.class.module_parent_name.underscore.dasherize
```

**Review by byroot — APPROVED** [review]
One nitpick, but LGTM.

**Comment by byroot** [comment]
Oh and a single commit please.

**Comment by st0012** [comment]
Out of curiosity, when can't PRs be squash-merged? And wouldn't it be better to mention the single commit requirement as a rule in the PR template?

**Comment by byroot** [comment]
> mention the single commit requirement as a rule in the PR template?

Actually there is no such thing as a single commit rule. It's perfectly fine for a PR to have as many commit as it needs. 

The reason we ask to squash is to have a somewhat clean git history, so commit messages such as "Update CHANGELOG" is something we try to avoid in the tree. It's way better when git blaming to try to find the context of a change.

As for not squash-merging, it's because the resulting squash commit generated by GitHub isn't great (or wasn't maybe it could be reconsidered, dunno).

**Comment by st0012** [comment]
>  It's perfectly fine for a PR to have as many commit as it needs.

Yeah that's my understanding too.

> It's way better when git blaming to try to find the context of a change.
> it's because the resulting squash commit generated by GitHub isn't great

That's fair but I don't think GH's generated message is much different than the manually squashed one.

For example, this is the final commit from this PR:

Screenshot 2024-01-20 at 23 43 36

And let's use this [IRB PR](https://github.com/ruby/irb/pull/835)'s (with 3 commits) squash-merge commit as another example:

Screenshot 2024-01-20 at 23 46 27

It similarly captures all 3 commit messages in the final commit. And even though the merge button was pushed by another maintainer, I'm still the author of the commit. So it shouldn't affect the contributors calculation either. Finally, you can edit the generated message before confirming the merge.

Anyway, I do squash merge in almost all my projects now and it does make maintenance work a bit easier.

**Comment by byroot** [comment]
I'd argue that the last two lines of that squashed commit shouldn't be there. But anyway, no big deal.

**Comment by byroot** [comment]
I also didn't double check your squashed commit. IMO the "Update changelog" part makes no sense.

When we ask for squashing, it's meant as: "Please have a single commit with a clean commit message". "Update changelog" isn't an interesting thing to add to a commit message. I can see from the git diff that the changelog is updated. So it's superfluous.

**Pull request merged**
